### DISCUSSION TITLE
Alias for Union[T, UndefinedType]

### DISCUSSION THREAD

**Discussion by pchanial** [wyfo/apischema#122] [Ideas]
Wouldn't NotRequired[T] have the sought after expressiveness?

**Comment by wyfo** [comment]
Yes it has, and I admit that `Union[T, UndefinedType]` could be a pretty common thing, and thus could have an alias in *apischema*.

Unfortunately, `NotRequired` is about to be "reserved" by [PEP 655](https://www.python.org/dev/peps/pep-0655), so I've decided to not use it, and because I haven't found an other suitable name, I've left things as they are for the moment; it's also simple enough to be written directly by people needing it in their own project.

Moreover, with [PEP 604](https://www.python.org/dev/peps/pep-0604/) coming in 3.10, this alias should become less useful, that's why I haven't thought too much about it …

So, I'm open to add an alias, but because I don't want to overlap with a PEP, it should be something else than `NotRequired`.
